test/inference/emcee.jl

```diff
@@ -0,0 +1,20 @@
+using Turing, Random, Test
+import Turing.Inference
+import AdvancedMH
+
+dir = splitdir(splitdir(pathof(Turing))[1])[1]
+include(dir*"/test/test_utils/AllUtils.jl")
+
+@testset "emcee.jl" begin
+    @testset "gdemo" begin
+        Random.seed!(9876)
+
+        n_samples = 1000
+        n_walkers = 250
+        
+        spl = Emcee(n_walkers, 2.0)
+        chain = sample(gdemo_default, spl, n_samples)
+
+        check_gdemo(chain[div(n_samples, 2):end])
```

Comment:
Maybe just run it for a bit longer or adjust the tolerance instead?

Reply:
Well, the burn-in on Emcee is quite extreme, and I think it's probably good modelling practice to remove the bulk of your early samples.

Reply:
I agree in practice, I was just a bit surprised about doing it in the tests since the tests in AdvancedMH are fine without discarding the warm-up phase.

Reply:
Turns out not to matter so much. I've adjusted the test.